polarify/main.py

```diff
@@ -139,25 +186,104 @@ class State:
 
     node: UnresolvedState | ReturnState | ConditionalState
 
+    def translate_match(
+        self,
+        subj: ast.expr | Sequence[ast.expr] | ast.Tuple,
+        stmt: ast.pattern,
+        guard: ast.expr | None = None,
+    ):
+        if isinstance(stmt, ast.MatchValue) and isinstance(subj, ast.Name):
+            if guard is not None:
+                return ast.BinOp(
+                    left=guard,
+                    op=ast.BitAnd(),
+                    right=ast.Compare(
+                        left=ast.Name(id=subj.id, ctx=ast.Load()),
+                        ops=[ast.Eq()],
+                        comparators=[stmt.value],
+                    ),
+                )
+
+            return ast.Compare(
+                left=ast.Name(id=subj.id, ctx=ast.Load()),
+                ops=[ast.Eq()],
+                comparators=[stmt.value],
+            )
+        elif isinstance(stmt, ast.MatchValue) and isinstance(subj, ast.Tuple):
+            return self.translate_match(subj, ast.MatchSequence(patterns=[stmt]))
+        elif isinstance(stmt, ast.MatchAs) and isinstance(subj, ast.Name):
+            if stmt.name is not None:
+                self.handle_assign(
```

Comment:
Is there any valid code that requires this?

Reply:
I was thinking of something similar to below:
```py
def match_x(x):
    match x:
        case _ if x > 5:
            return 0
        case _:
            return 1
```
In which case `ast` returns an empty `MatchAs` in which case the `if` statement on line 215 is required.